src/entry-client.js

```diff
@@ -114,130 +111,108 @@ export function perfMeasure(name) {
     return true;
 }
 
-export default function initializeClient(createApp, clientOpts) {
-    const opts = {
-        appSelector: '#app',
-        hmr: true,
-        initialState: null,
-        initialStateMetaTag: 'initial-state',
-        vuexModules: true,
-        middleware: () => Promise.resolve(),
-        globalFetchData: () => Promise.resolve(),
-        postMiddleware: () => Promise.resolve(),
-        logger: console,
-        enablePerfMarks: false,
-        // By default, only run fetchData middlewares on path changes
-        shouldProcessRouteUpdateDefaults: {
-            path: true,
-            query: false,
-            hash: false,
-        },
-        ...clientOpts,
-    };
-
-    // Store off for closure scope usage
-    enablePerfMarks = opts.enablePerfMarks;
-
-    const { shouldProcessRouteUpdateDefaults: spruDefaults } = opts;
-    let { initialState } = opts;
+/**
+ * Register/Unregister any dynamic Vuex modules during client-side routing operations.
+ * Registering the store modules as part of the component allows the module to be bundled
+ * with the async-loaded component and not in the initial root store bundle
+ *
+ * @param   {object} app Destination route object
+ * @param   {object} router Destination route object
+ * @param   {object} store  Vuex Store instance
+ * @param   {object} logger Logger instance
+ * @returns {undefined}     No return value
+ */
+export function useRouteVuexModulesClient(app, router, store, logger) {
```

Comment:
Expose this new composable utility to wire up route level vuex module handling - logic remains the same thought with minor cleanups.